### DISCUSSION TITLE
What is the purpose of the table [latest_metrics] in mlflow DB?

### DISCUSSION THREAD

**Discussion by arees88** [mlflow/mlflow#7063] [Q&A] [answered]
Hi,

**Re: Purpose of `latest_metrics` table in mlflow DB**

In the initial version of MLflow only the `metrics` table was present. 

Does anybody know why the `latest_metrics` table was introduced and what it's purpose / function is, please?

Many thanks in advance.

A Rees

**Comment by JHibbard** [comment, marked as answer]
Hi Rees,

The `latest_metrics` table is a [feature introduced](https://github.com/mlflow/mlflow/pull/1767) to improve performance:

> Create a materialized metrics table (latest_metrics) that contains the summary value for each unique (metric_key, run) tuple. When a run is fetched via the GetRun API, this summary value should be used when populating the run's data field.
>
> The addition of this table will remove the need for a complex query that fetches the most recent value for a particular (metric_key, run) tuple based on step, timestamp, and value, thus providing performance improvements.
>
> This PR also introduces eager loading for run attributes when appropriate (e.g., during search_runs and get_runs invocations); this helps avoid repeated database accesses that occur when model object attributes are accessed under a lazy load paradigm.

Hope that helps!
Jim

**Reply by arees88** [reply]
Yes it does help. 
Many thanks for the prompt reply, @JHibbard.

Regards,
Agnieszka
